app/src/main/java/com/rkm/tasky/feature/agenda/screen/dayselector/DateSelectorStateImpl.kt

```diff
@@ -0,0 +1,74 @@
+package com.rkm.tasky.feature.agenda.screen.dayselector
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableLongStateOf
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.listSaver
+import androidx.compose.runtime.saveable.rememberSaveable
+
+@Stable
+internal class DateSelectorStateImpl(
+    initialSelectedDate: Long,
+    private val timeSpan: TimeSpan,
+) : DateSelectorState {
```

Comment:
This class looks to manage the calculation of the `x` amount of days after a selected date. I believe your implementation here works, but I’d like to share another approach that you may find interesting:

```
class AgendaOverviewViewModel(…) {
  private var startDate = MutableStateFlow<LocalDate>(LocalDate.now())
  private val rangeOfDates = MutableStateFlow<List<LocalDate>>(emptyList())
  private var selectedDate = MutableStateFlow<LocalDate>(LocalDate.now())

  init {
    combine(
      startDate,
      selectedDate,
    ) { startDate, selectedDate ->
    
      if (startDate != rangeOfDates.value.firstOrNull()) {
        val dates = generateDays(
          startDate = startDate,
          numberOfDays = 6,
        )
        rangeOfDates.value = dates
        this.selectedDate.value = startDate
      }
    
      // … update your state
    
    }.launchIn(viewModelScope)
  }

private fun generateDays(
  startDate: LocalDate,
  numberOfDays: Int,
): List<LocalDate> {
  return List(numberOfDays) {
    startDate.plusDays(it.toLong())
  }
}
```